spec/requests/api/v1/custom_emojis_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'Custom Emojis' do
+  let(:user)    { Fabricate(:user) }
+  let(:token)   { Fabricate(:accessible_access_token, resource_owner_id: user.id) }
+  let(:headers) { { 'Authorization' => "Bearer #{token.token}" } }
+
+  describe 'GET /api/v1/custom_emojis' do
+    before do
+      Fabricate(:custom_emoji, domain: nil, disabled: false, visible_in_picker: true)
+    end
+
+    context 'when not authorized' do
```

Comment:
This is confusing, you're returning data “when not authorized”. What it tests is that it returns data when logged out:
```suggestion
    context 'when logged out' do
```

There is a case where someone logged out is legitimately *unauthorized*, it's when unauthenticated API access is disabled (`DISALLOW_UNAUTHENTICATED_API_ACCESS=true`), but that is not what was being tested.